### ISSUE TITLE
RuntimeError: can't modify frozen ActiveSupport::HashWithIndifferentAccess

### ISSUE THREAD

**Issue by hodak** [zacheryph/hstore_flags#1]
Setter raises this error on Rails 4.2.0

```ruby
u = User.new
u.active
#=> false
u.active = true
RuntimeError: can't modify frozen ActiveSupport::HashWithIndifferentAccess
from /Users/hodak/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-4.2.0/lib/active_support/hash_with_indifferent_access.rb:135:in `[]='
```

When you google this problem it seems like other libraries had problems with this update too.

**Comment by zacheryph** [comment]
I will take a look at this tonight and get a fix out there

**Comment by aligo** [comment]
Confirmed that on rails 4.2
For now I just simply drop dirty tracking for flag property.
https://github.com/aligo/hstore_flags/commit/de7d27cfbe2f2087779fe59fb8303cf22bdf0494

**Comment by svoop** [comment]
@zacheryph Any news on this?

**Comment by svoop** [comment]
@aligo Thanks for your quick fix, unfortunately, it doesn't work when flags are updated as follows:

    # in User.rb: hstore_flags :welcome_mail_sent
    user.update_attribute :welcome_mail_sent, true
    # => can't modify frozen ActiveSupport::HashWithIndifferentAccess

**Comment by svoop** [comment]
@zacheryph See #2 for a solution. Not particularly beautiful, but since the changed_attributes hash is frozen by AR, I don't see another way. Thanks for taking a look and considering to merge and release this fix.

**Comment by zacheryph** [comment]
Yeah so I totally forgot about this... :cry:

I have updated and released v0.4.1 with the thanks of @svoop

**Comment by svoop** [comment]
@zacheryph Great, thanks a bunch!
